src/iam/iam-base.ts

```diff
@@ -673,4 +677,31 @@ export class IAMBase {
       })
     });
   }
+
+  private async initContracts() {
+    const { chainId } = await this._provider.getNetwork();
+    const {
+      ensRegistryAddress, ensResolverAddress, didContractAddress, cacheServerUrl
+    } = IAMBase.chainConfigs[chainId];
+    this._registrySetting = {
+      address: didContractAddress,
+      abi: abi1056,
+      method: Methods.Erc1056
+    };
+    this._ensRegistryAddress = ensRegistryAddress;
+    this._ensResolverAddress = ensResolverAddress;
+    this._ensRegistry = EnsRegistryFactory.connect(ensRegistryAddress, this._provider);
+    this._ensResolver = PublicResolverFactory.connect(ensResolverAddress, this._provider);
+    cacheServerUrl && (this._cacheClient = new CacheServerClient({ url: cacheServerUrl }));
+  }
+
+  /**
+   * @description - should be invoked before iniitialization
+   * 
+   * @param chainId 
+   * @param config 
+   */
+  static setChainConfig(chainId: number, config: ChainConfig) {
```

Comment:
Cool approach 🧠 . Just curious, why this approach rather than passing into the constructor? I think it could be nice, it just deviates a bit from the "config in the constructor" approach that we have so far?

Reply:
Since there are just a few possible chain configurations its make sense to set them as defaults. `setChainConfig` is supposed to be used to override or add missing configuration. Perhaps some documentation will be useful